tests/baselines/reference/nodeModulesImportTypeModeDeclarationEmitErrors1(module=node16).js

```diff
@@ -121,16 +121,16 @@ exports.b = null;
 
 //// [index.d.ts]
 export type LocalInterface = import("pkg", { assert: { "resolution-mode": "foobar" } }).RequireInterface & import("pkg", { assert: { "resolution-mode": "import" } }).ImportInterface;
-export declare const a: import("pkg").RequireInterface;
-export declare const b: import("pkg", { with: { "resolution-mode": "import" } }).ImportInterface;
+export declare const a: import("pkg", { assert: { "resolution-mode": "foobar" } }).RequireInterface;
+export declare const b: import("pkg", { assert: { "resolution-mode": "import" } }).ImportInterface;

```

Comment:
I think we're trying to avoid emitting `assert`; is this desirable?

Reply:
I could change the reuse logic to never reuse types that have `assert` in them and use the printer instead.

Reply:
Changed to not reuse import types with assert keyword.